packages/@glimmer/compiler/lib/passes/1-normalization/visitors/strict-mode.ts

```diff
@@ -0,0 +1,380 @@
+
+      case 'InterpolateExpression':
+        return this.InterpolateExpression(expression, span, resolution);
+
+      case 'CallExpression':
+        return this.CallExpression(expression, span, resolution ?? ResolutionType.Helper);
+
+      case 'Not':
+        return this.Expression(expression.value, span, resolution);
+
+      case 'IfInline':
+        return this.IfInline(expression);
+
+      case 'Curry':
+        return this.Curry(expression);
+
+      case 'Log':
+        return this.Log(expression);
+    }
+  }
+
+  Args(args: mir.Args): Result<null> {
+    return this.Positional(args.positional)
+      .andThen(() => this.NamedArguments(args.named));
+  }
+
+  Positional(positional: mir.Positional, span?: HasSourceSpan): Result<null> {
+    let result = Ok(null);
+    let expressions = positional.list.toArray();
+
+    // For cases like {{yield foo}}, when there is only a single argument, it
+    // makes for a slightly better error to report that entire span. However,
+    // when there are more than one, we need to be specific.
+    if (expressions.length === 1) {
+      result = this.Expression(expressions[0]!, span);
+    } else {
+      result = this.Expressions(expressions);
+    }
+
+    return result;
+  }
+
+  NamedArguments({ entries }: mir.NamedArguments): Result<null> {
+    let result = Ok(null);
+
+    for (let arg of entries.toArray()) {
+      result = result.andThen(() => this.NamedArgument(arg));
+    }
+
+    return result;
+  }
+
+  NamedArgument(arg: mir.NamedArgument): Result<null> {
+    if (arg.value.type === 'CallExpression') {
+      return this.Expression(arg.value, arg, ResolutionType.Helper);
+    } else {
+      return this.Expression(arg.value, arg);
+    }
+  }
+
+  ElementParameters({ body }: mir.ElementParameters): Result<null> {
+    let result = Ok(null);
+
+    for (let param of body.toArray()) {
+      result = result.andThen(() => this.ElementParameter(param));
+    }
+
+    return result;
+  }
+
+  ElementParameter(param: mir.ElementParameter): Result<null> {
+    switch (param.type) {
+      case "StaticAttr":
+        return Ok(null);
+      case "DynamicAttr":
+        return this.DynamicAttr(param);
+      case "Modifier":
+        return this.Modifier(param);
+      case "SplatAttr":
+        return Ok(null);
+    }
+  }
+
+  DynamicAttr(attr: mir.DynamicAttr): Result<null> {
+    if (attr.value.type === 'CallExpression') {
+      return this.Expression(attr.value, attr, ResolutionType.Helper);
+    } else {
+      return this.Expression(attr.value, attr);
+    }
+  }
+
+  Modifier(modifier: mir.Modifier): Result<null> {
+    return this.Expression(modifier.callee, modifier, ResolutionType.Modifier)
+      .andThen(() => this.Args(modifier.args));
+  }
+
+  InElement(inElement: mir.InElement): Result<null> {
+    return this.Expression(inElement.destination)
+      // Unfortunately we lost the `insertBefore=` part of the span
```

Comment:
There is no AST/mir node (and hence span) for the “name=” part of named arguments; there is AST/mir node for the entire “name=value” thing, but the InElement mir node did not preserve it since it happen to know it will always have (or add one during normalization) the insertBefore key. So we just have the expression node and the entry node was thrown away in earlier normalization pass